packages/netlify-cms-core/src/actions/config.js

```diff
@@ -63,6 +68,8 @@ export function applyDefaults(config) {
         map.setIn(['slug', 'sanitize_replacement'], '-');
       }
 
+      const backend = map.getIn(['backend', 'name']);
```

Comment:
> This will prevent mistakes of using a default locale that is not in the top level list:

The `locales` and `default_locale` are merged together and set up as collection level `locales`.

Fixed the gif issue.